ethclient/ethclient.go

```diff
@@ -257,6 +257,14 @@ func toBlockNumArg(number *big.Int) string {
 	if number == nil {
 		return "latest"
 	}
+	switch number.Int64() {
+	case rpc.PendingBlockNumber.Int64():
```

Comment:
1. There is potential loss of information here since `big.Int` might not be `Int64`.
2. See the following code:
```go
// eth/api.go

func (api *PrivateDebugAPI) TraceBlockByNumber(blockNr rpc.BlockNumber, config *vm.LogConfig) BlockTraceResult {
	// Fetch the block that we aim to reprocess
	var block *types.Block
	switch blockNr {
	case rpc.PendingBlockNumber:
		// Pending block is only known by the miner
		block = api.eth.miner.PendingBlock()
	case rpc.LatestBlockNumber:
		block = api.eth.blockchain.CurrentBlock()
	default:
		block = api.eth.blockchain.GetBlockByNumber(uint64(blockNr))
	}
```
```go
// rpc.types.go

type BlockNumber int64

const (
	PendingBlockNumber  = BlockNumber(-2)
	LatestBlockNumber   = BlockNumber(-1)
	EarliestBlockNumber = BlockNumber(0)
)
```
Why not use the original types instead of casting? This feels somewhat hacky...

Reply:
1. you're right, number.Int64() might return undefined result if cannot be converted
2. the issue is that the defined consts rpc.types are basically int64, while the input in mobile case is big.Int. so some conversion needed, I can create a big.Int for each case like the following, but that would be wasteful allocations, I might also define the mobile consts as bin.Int for the first place. but they cannot be consts that way.

```	
        switch number {
	case big.NewInt(rpc.PendingBlockNumber.Int64()):
		return "pending"
	case big.NewInt(rpc.LatestBlockNumber.Int64()):
		return "latest"
	case big.NewInt(rpc.EarliestBlockNumber.Int64()):
		return "earliest"
	}